File: CSE241/HW7/driver2.java
import java.util.Scanner;

public class driver2{

    /**
     * Main method for driver2. Takes input from the user.
     * @param args
     */
    public static void main(String[] args) {
        Scanner s = new Scanner(System.in);
        int row = 1;
        int col = 1;
        try{
            System.out.println("Please enter the size of the tetris board");
            try {
                row = s.nextInt();
                col = s.nextInt();
                if(row < 0 || col < 0 ){
                    throw new IllegalArgumentException("Row or column cannot be negative integers!");
                }    
            } catch (IllegalArgumentException e){
                System.out.println(e.getMessage());
            }
            Tetris board = new Tetris(row, col);
            
            while(true){
            System.out.println("Please enter the tetromino type. R for random. Q to quit.");
            char p = s.next().charAt(0);
            while(true){
                if(p == 'I' || p == 'O' || p == 'T' || p == 'L' || p == 'J' || p == 'S' || p == 'Z' || p == 'R' || p == 'Q'){
                    break;
                }
                else{
                    System.out.println("Please enter valid tetromino type!");
                    p = s.next().charAt(0);
                }
            }
            Tetromino.Piece piece = Tetromino.Piece.O;
            switch(p){
                case 'I':
                    piece = Tetromino.Piece.I;
                    break;
                case 'O':
                    piece = Tetromino.Piece.O;
                    break;
                case 'T':
                    piece = Tetromino.Piece.T;
                    break;
                case 'J':
                    piece = Tetromino.Piece.J;
                    break;
                case 'L':
                    piece = Tetromino.Piece.L;
                    break;
                case 'S':
                    piece = Tetromino.Piece.S;
                    break;
                case 'Z':
                    piece = Tetromino.Piece.Z;
                    break;
                case 'R':
                    piece = Tetromino.Piece.R;
                    break;
                case 'Q':
                    piece = Tetromino.Piece.Q;
                    break;
                default:
                    System.err.println("Please enter valid tetromino type");
                    break;
            }
    
            if(piece == Tetromino.Piece.Q){
                System.out.println("Exiting.");
                break;
            }
                Tetromino t1 = new Tetromino(piece);
                t1.print();
                board.add(t1);
                board.draw();
                board.animate(t1);
            }
        } finally{
            s.close();
        }

    }
}

File: CSE241/HW2/driver2.cpp
#include <iostream>
#include <vector>

#include "tetris.hpp"
#include "tetromino.hpp"

using namespace std;

int main(){
    int row_size;
    int col_size;

    while (1){ // Valid entry check
        cout << "Please enter the row and columnn of the tetris board:" << endl;        
        if (cin >> row_size >> col_size && row_size > 0 && col_size > 0){ 
            break;
        }
        else{
            cout << "Illegal value. Please enter a valid value!" << endl;
            cin.clear();
            cin.ignore(31, '\n');
        }
    }
    Tetris tetris(row_size,col_size);
    tetris.createBoard();
    do{
        char p;
        int i = 0;
        while(1){ // Valid tetromino type entry check
            cout << "Enter the tetromino type.'R' for random and 'Q' for quit" << endl;
            if(cin >> p && ((p == 'I') ||(p == 'O') ||(p == 'T') ||(p == 'L') ||(p == 'J') ||(p == 'S') ||(p == 'Z') ||(p == 'R') || (p == 'q') || (p == 'Q'))){
                break;
            }
            else{
                cout << "Illegal tetromino type. Please enter a valid type!" << endl;
                cin.clear();
                cin.ignore(31,'\n');
            }
        }
        vector<Tetromino> T;
        T.push_back(static_cast<Piece>(p));
        if(p == 'q' || p == 'Q'){
            cout << "Exiting..." << endl;
            break;
        }
        
        tetris.Add(T[i].getShape());
        if(tetris.Fit()){ 
            tetris.Animate(T[i].getShape());
        }
        else{
            T[i].rotate('l');
            tetris.Animate(T[i].getShape());
        }
        i++;
    }while (1);
    
    return 0;
}